packages/react-scripts/template/README.md

```diff
@@ -901,6 +902,92 @@ In the future we plan to integrate it into Create React App even more closely.
 
 To learn more about Flow, check out [its documentation](https://flowtype.org/).
 
+## Adding TypeScript
+
+TypeScript is a typed superset of JavaScript that compiles to plain JavaScript.
+
+Recent versions of [TypeScript](https://www.typescriptlang.org/) work with Create React App projects out of the box thanks to Babel 7. Beware that Babel 7 TypeScript does not allow some features of TypeScript such as constant enum and namespaces.
+
+To add TypeScript to a Create React App project, follow these steps:
+
+1. Run `npm install --dev @types/react @types/react-dom` (or `yarn add --dev @types/react @types/react-dom`).
+2. Rename all `.js` files to `.tsx` if they have React components or `.ts` if not (e.g. `git mv src/index.js src/index.tsx`).
+3. Create a `tsconfig.json` file with this content:
+
+```json
+{
+  "compilerOptions": {
```

Comment:
May i propose a simpler `tsconfig.json`? I know for example recommending all of the `strict*` options may discourage the CRA team from considering this (see https://github.com/wmonk/create-react-app-typescript/issues/333 for example).

Something like:

```js
    "compilerOptions": {
        "target": "esnext",
        "module": "esnext",
        "moduleResolution": "node",
        // Note that including es2017 would include APIs not available in older browsers (IE11)
        "lib": ["es5", "es2015.promise", "es2018.promise", "dom"],
        "allowJs": true,
        "skipLibCheck": true,
        "jsx": "react",
        "allowSyntheticDefaultImports": true,
        "noEmit": true,
        // Emits nicer type checking output
        "pretty": true

        // I'm guessing this isn't needed since TS isn't actually used
        // "sourceMap": true,

        // Perhaps link to TS compiler options to suggest reading to enab
        // "strict": true, <-- strict enables a bunch of the following options also
        // "alwaysStrict": true,
        // "strictNullChecks": true,
        // "noImplicitAny": true,
        // "strictFunctionTypes": true,
        // "noUnusedLocals": true,
        // "noUnusedParameters": true,
        // "noImplicitReturns": true,
        // "noFallthroughCasesInSwitch": true,
        // "noImplicitThis": true,

        // I don't think CRA supports this style of absolute imports. Maybe i'm wrong?
        // "baseUrl": "./",
        // "paths": {
        //     "*": ["*.web", "*"]
        // },

        // I don't believe CRA supports decorators
        // "experimentalDecorators": true,
        // "emitDecoratorMetadata": true
    }
```

Reply:
The `pretty` option thing may not have an effect here since we're using `ForkTsCheckerWebpackPlugin`.

Reply:
`pretty` defaults to true anyway since typescript 2.9 [see here](https://github.com/Microsoft/TypeScript/wiki/What%27s-new-in-TypeScript#--pretty-output-by-default)